src/Faker/Provider/zh_CN/Text.php

```diff
@@ -0,0 +1,1063 @@
+        $insertPunctuation = array();
+        $lastInsertPos = 0;
+        $lastCharIsPunctuation = '';
+
+        foreach ($words as $pos => $word) {
+            if (array_key_exists($word, static::$punctuationPairs)) {
+                $expectedPunctuation[] = static::$punctuationPairs[$word];
+                $expectedPunctuationInsertPos[] = -1;
+            } elseif (in_array($word, static::$punctuationPairs)) {
+                if (!empty($expectedPunctuation) && $word === $expectedPunctuation[count($expectedPunctuation) - 1]) {
+                    array_pop($expectedPunctuation);
+                    array_pop($expectedPunctuationInsertPos);
+                } else {
+                    $insertPunctuation[$lastInsertPos] = array_search($word, static::$punctuationPairs);
+                }
+            } elseif (in_array($word, static::$punctuations)) {
+                if (!$lastCharIsPunctuation) {
+                    if (!empty($expectedPunctuationInsertPos) && $expectedPunctuationInsertPos[count($expectedPunctuationInsertPos) - 1] < 0) {
+                        $expectedPunctuationInsertPos[count($expectedPunctuationInsertPos) - 1] = $pos;
+                    }
+                }
+                $lastCharIsPunctuation = true;
+            } else {
+                if ($lastCharIsPunctuation) {
+                    $lastInsertPos = $pos;
+                }
+                $lastCharIsPunctuation = false;
+            }
+        }
+
+        foreach ($expectedPunctuationInsertPos as $index => $pos) {
+            $insertPunctuation[$pos] = $expectedPunctuation[$index];
+        }
+
+        foreach ($insertPunctuation as $pos => $punctuation) {
+            $words[$pos] = $punctuation . $words[$pos];
+        }
+
+        return static::implode($words);
+    }
+
+    /**
+     * UTF-8 strlen
+     *
+     * @see https://en.wikipedia.org/wiki/UTF-8#Description
+     *
+     * @param string $text
+     *
+     * @return int
+     */
+    protected static function utf8Strlen($text)
+    {
+        $count = 0;
+        for ($i = 0, $len = strlen($text); $i < $len; $i++) {
+            $char = ord($text[$i]);
+            if ($char <= 0x7F) {
+                // 00000000 - 01111111
+                // ASCII
+                $count++;
+            } elseif ($char <= 0xBF) {
+                // 10000000 - 10111111
+                // Not the first byte. Do nothing.
+            } elseif ($char <= 0xFD) {
+                // 11000000 - 11011111
+                // 11100000 - 11101111
+                // 11110000 - 11110111
+                // 11111000 - 11111011
+                // 11111100 - 11111101
+                // The first byte of UTF-8 character
+                $count++;
+            } else {
+                // Unexpected first byte of UTF-8 character
+            }
+        }
+        return $count;
+    }
+
+    /**
+     * Get UTF-8 string characters array
+     *
+     * TODO: this function is not used
```

Comment:
Why is this method in here when it is not used? And in general speaking i dont think such a method even belongs in a locale. If it belongs somewhere then it would even be in the core itself

Reply:
This function `utf8Chars` is a fallback for `explode`. I don't know if `preg_split` with `/u` works well for utf8 text. So I provide a fallback function. But it's truly not used.